src/App.js

```diff
@@ -1,40 +1,78 @@
-import React, { useState, useEffect } from 'react';
-import { SearchBar } from './modules/recipes/search-bar';
-import { fetchRecipesByIngredient, fetchRecipeDetailsById } from './modules/recipes/recipeService';
-import './App.css';
-import { RecipeList } from './modules/recipes/recipe-list/RecipeList';
-import { RecipeDetail } from './modules/recipes/recipe-detail/RecipeDetail.js'; 
+import { useState } from "react";
+import { BrowserRouter as Router, Routes, Route } from "react-router-dom";
+import { UserProvider } from "./Contexts/UserContext";
+import { NavBar } from "./shared-components/NavBar";
+import { SearchBar } from "./modules/recipes/SearchBar";
+import { RecipeList } from "./modules/recipes/RecipeList";
+import { RecipeDetails } from "./modules/recipes/RecipeDetails";
+import { Categories } from "./modules/recipes/Categories";
+import { CategoryDetails } from "./modules/recipes/CategoryDetails";
+import { CreateRecipe } from "./modules/recipes/CreateRecipe";
+import { ProtectedRoute } from "./ProtectedRoute";
+import { fetchRecipeDetailsById } from "./modules/recipes/recipeService";
+import { useEffect } from 'react';
+import "./App.css";
 
-export const App = () => {
-    const [recipes, setRecipes] = useState([]);
-    const [selectedRecipe, setSelectedRecipe] = useState(null); 
+export const App = ({ recipes, onRecipeClick }) => {
+  const [selectedRecipe, setSelectedRecipe] = useState(null);
+  const [isLoading, setIsLoading] = useState(true);
 
-    useEffect(() => {
-        fetchRecipesByIngredient('chicken_breast').then((recipes) =>
-            setRecipes(recipes.length > 0 ? recipes : []),
-        );
-    }, []);
+  useEffect(() => {
+    // API çağrısı burada yapılabilir
+    // Veri yüklendiğinde setIsLoading(false) yapın
+  }, []);
 
-    const handleRecipeClick = async (idMeal) => {
-        const details = await fetchRecipeDetailsById(idMeal);
-        setSelectedRecipe(details);
-    };
+  if (isLoading) {
+    return <div>Loading...</div>;
+  }
 
-    return (
-        <div className='container'>
-            <header>Recipe Search App</header>
-            <SearchBar />
-            {recipes.length > 0 ? (
-                <RecipeList recipes={recipes} onRecipeClick={handleRecipeClick} />
-            ) : (
-                <></>
-            )}
-            {selectedRecipe && <RecipeDetail recipe={selectedRecipe} />} {/* Detaylı tarif bileşenini göster */}
+  const handleRecipeClick = async (idMeal) => {
+    const details = await fetchRecipeDetailsById(idMeal);
+    setSelectedRecipe(details);
+    if (onRecipeClick) {
+      onRecipeClick(details);
+    }
+  };
+
+  return (
+    <UserProvider>
+      <Router>
```

Comment:
All routes should be defined in the AppRouter.js file as we did in the lesson, do not define and routes outside  AppRouter.js file.